### PR TITLE
Draggable queue items

### LINKED ISSUE #332: Add draggable list component to the Queue page
The Queue page is currently complete in v4 _except_ for drag-and-drop reordering of queue items.

image

For this issue, we need to wrap the existing Queue page list component with a 3rd party draggable list component.

It's a pretty big task, and if someone could even make partial progress on adding the component, we could continue it from there.

Also, feel free to skip handling the network requests to set the new queue order on the server. Otherwise [here is an example](https://github.com/podverse/podverse-rn/blob/develop/src/services/queue.ts#L87-L120) of how queue items are reordered in the Podverse mobile app. You don't need to include the bearerToken or Locally handling in the web version.

react-draggable-list seems like the currently most used React npm library used for this purpose: https://www.npmjs.com/package/react-draggable-list

### PR DESCRIPTION
**Pull request by kbondarev**
Closes #332 

This is my first time contributing, so any feedback is welcome regarding style guide or anything else :smile: 

Regarding the UI, there are a few options:
1. Hide the drag handle/button by default. Show it only when the **edit button** is clicked together with the remove buttons (_current behavior_ :heavy_check_mark: )
    PV_Queue_1
3. Show the drag handle by default (on the right side). The **edit button** will toggle between the remove and drag buttons (similar to how it is in the mobile app)
    PV_Queue_2
4. Something completely different? :thinking:

### CODE REVIEW

**Comment by mitchdowney** [comment]
@kbondarev thanks so much for working on this! It has been a busy holiday weekend for me, but I'm free most of tomorrow and should have a chance to review and test this then.

File: src/components/ClipListItem/ClipListItem.tsx [hunk 1]
```diff
@@ -119,23 +122,35 @@ export const ClipListItem = ({
...
           {isChapter && showImage && finalImageUrl && (
             <PVImage alt='' height={PV.Images.sizes.medium} src={finalImageUrl} width={PV.Images.sizes.medium} />
           )}
           {itemMainSection}
         </div>
-        {showRemoveButton && (
+        {(showRemoveButton) && (
           <div className='side-wrapper'>
-            <ButtonCircle
+            {showRemoveButton && <ButtonCircle
```
Comment (kreonjr) [inline src/components/ClipListItem/ClipListItem.tsx:145]: If this is checked above (on line 143), does it need to be checked again here?

**Review by kreonjr — APPROVED** [review]
Tiny comment for a redundant check but still approved :)

**Review by mitchdowney — APPROVED** [review]
It works perfect! I'll make the tiny code change suggestion Creon had, then deploy into prod shortly.

**Comment by mitchdowney** [comment]
@kbondarev thanks so much for your help! This has been a long overdue feature and I know people will appreciate it.

File: src/components/ClipListItem/ClipListItem.tsx (hunk 1 above)
Comment (kbondarev) [reply src/components/ClipListItem/ClipListItem.tsx:145]: You're right. I had the drag button on the right-side first, and therefore had those extra checks bout in the wrapping div, and around each button. But I guess I forgot to clean it up once I moved the drag button to the left side

**Pull request merged**